### PR TITLE
Example for budgeting methods

### PR DESCRIPTION
**Pull request by fpiwowarczyk**
Here is first example into our repository. 

I tried to describe budgeting methods as well as I could. My goal was to show how to prepare SLO with any budgetingMethod and show differences between them.

If you see anything that should be made different or could be done better feels free to point it here.

### CODE REVIEW

File: examples/budgetting-method/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Budgeting method
+
+OpenSLO specifies three types of budgeting methods:
+
+- Occurrences
+- Timeslices
+- RatioTimeslices
+
+Here is bries description of them and examples of use cases for each method.
```
Comment (kenfinnigan) [inline examples/budgetting-method/README.md:9]: s/bries/a brief

File: examples/budgetting-method/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+How would DDoS attack impact your service for different budgeting methods.
+
+Assumptions:
+
+- Timewindow: 2 weeks rolling
+- Average daily requests: 24000
+- Your service by average works with: 99% of success responses.
+- Objective is: 95% of successful requests.
+- Timeslice is 1minute (if it exists).
+- By request I mean here anything that can end with success or failure please put in here your SLI it is just easier for calculations.
```
Comment (kenfinnigan) [inline examples/budgetting-method/README.md:48]: Should this be two sentences? Reading it I'm not entirely sure what it's trying to say

File: examples/budgetting-method/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Budgeting method
+
+OpenSLO specifies three types of budgeting methods:
```
Comment (kenfinnigan) [inline examples/budgetting-method/README.md:3]: Could the folder be renamed to have a single "t" in "budgeting" to align with the text?

File: examples/budgetting-method/README.md (hunk 1 above)
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md:9]: Thanks, typo, fixed :bowing_man:

File: examples/budgetting-method/README.md (hunk 2 above)
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md:48]: I rephrased this sentence little bit. My goal was to show that it is example how some actions will impact error budget and i'm using some simplifications. Thanks!

File: examples/budgetting-method/README.md (hunk 3 above)
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md:3]: Yes! Sorry, for that forget to look on file names :D

File: examples/budgetting-method/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- Your service was under attack by a malicious group that decided to DDoS your service making it working with 1% of availability.
+
+#### Occurrences burned budget
+
+In two weeks you have 336000 requests. If a malicious group decides to attack your service with DDoS it will take 16632
+failed requests to burn your whole error budget. Depending on the attack it can take different time but shouldn't be long.
+
+#### Timeslices burned budget
+
+Two weeks timewindow have 20160 timeslices, to burn your budget attack need to last for 16 hours and 48 minutes.
```
[the remark this answers is not in the document]
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md]: Fixed

File: examples/budgetting-method/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+Assumptions:
+
+- Timewindow: 2 weeks rolling
+- Average daily requests: 24000
+- Your service by average works with: 99% of success responses.
+- Objective is: 95% of successful requests.
+- Timeslice is 1minute (if it exists).
+- By request I mean here anything that can end with success or failure please put in here your SLI it is just easier for calculations.
+- Your service was under attack by a malicious group that decided to DDoS your service making it working with 1% of availability.
```
[the remark this answers is not in the document]
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md]: Done

File: examples/budgetting-method/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+How would DDoS attack impact your service for different budgeting methods.
+
+Assumptions:
+
+- Timewindow: 2 weeks rolling
+- Average daily requests: 24000
+- Your service by average works with: 99% of success responses.
+- Objective is: 95% of successful requests.
+- Timeslice is 1minute (if it exists).
```
[the remark this answers is not in the document]
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md]: Yup

File: examples/budgetting-method/README.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+often says "Service needs to be available 99% of time". That can be understood as 99% of minutes have 90% of successful
+requests.
+
+## RatioTimeslices
+
+Method uses an average of all time slices' success ratios. Here as in timeslices method timewindow is divided into parts.
+Average of ratio of success to total counts determine if error budget is burned. This method is good when the monitored
+service is exposed to an unpredictable environment and needs to be safe from rare situations when the whole error budget
+could be burned in minutes. For example let's imagine some service that works well and has by average 1 million of requests
+weekly. If some malicious group decided to prepare a DDoS attack and overflow our service with thousands of requests for
```
[the remark this answers is not in the document]
Comment (fpiwowarczyk) [reply examples/budgetting-method/README.md]: It sounds much better, thanks for suggestion.

**Comment by weyert** [comment]
Bit of a selfish idea but I think it would be cool if a next step of this example is to somehow make the example work with Prometheus recording rules. I think it could be a building block to extend `slogen` in the future to automatically generate it

**Review by kenfinnigan — APPROVED** [review]

**Review by programmer04 — APPROVED** [review]

**Comment by fpiwowarczyk** [comment]
> Bit of a selfish idea but I think it would be cool if a next step of this example is to somehow make the example work with Prometheus recording rules. I think it could be a building block to extend `slogen` in the future to automatically generate it

@weyert Thanks for your suggestion as I understand this example is good but could be extended in separate pull requests. Could you create Issue with that idea?

**Comment by fpiwowarczyk** [comment]
This PR partially implements #174

**Pull request merged**